### DISCUSSION TITLE
مشکلات استفاده از MemoryCache  در production

### DISCUSSION THREAD

**Discussion by milad-i** [Sina-Soltani/Parbad#234] [Q&A]
<div dir="rtl">
سلام.<br>
اگر در production هم از MemoryCache  استفاده کنیم، چه مشکلاتی پیش میاد؟<br>
چون من MemoryCache   رو به راحتی راه اندازی کردم ولی EntityFramework رو نتونستم.
</div>

**Comment by Sina-Soltani** [comment]
درود
در صورت استفاده از مموری کش، اگر سرور و یا وبسایت شما ریستارت بشه، اطلاعات موجود در مموری از بین خواهد رفت.
اطلاعاتی که در این پایگاه داده (مموری کش در موقعیت شما) ذخیره شده، شامل کدهای رهگیری، وضعیت فعلی هر تراکنش و کلیه اطلاعات تراکنش ها هست.
بنابراین اگر کاربری درخواست پرداختی داده باشه و در صفحه بانک در حال پرداخت باشه، و سرور و یا وب سایت شما ریستارت بشه، زمانی که کاربر به وب سایت شما برمیگرده تا عملیات نهایی بشه، هیچ اطلاعاتی برای پردازش موجود نیست و پیغام متناسب با اون دریافت میکنید. البته مسلما هزینه به صورت خودکار توسط بانک به حساب کاربر برمیگرده.
مموری کش برای محیط تست و توسعه باید استفاده بشه نه محیط نهایی.
پیاده سازی EF Core بسیار راحته، اگر تجربه کار با اون رو داشته باشید.
نمونه پروژه هم موجوده.
